### ISSUE TITLE
Relative include paths aren't working

### ISSUE THREAD

**Issue by ICJefferson** [Southclaws/vscode-pawn#3]
I recently switched from Sublime Text 3 and all my PAWN projects are written in it. Your package for ST3 allowed relative paths in include directives:
```PAWN
#include "..\include\modules\core\test.pwn"
```
In VSCode, using this extension, it fails to find the file. Unless, I replace the '..' with its absolute equivalent:
```PAWN
#include "E:\Projects\SAMP\server-master\include\modules\core\test.pwn"
```

Including `.inc` files using the bracket notation works fine:
```PAWN
#include <a_samp> // is included as expected
```

My `tasks.json`:
```JSON
{
    "version": "2.0.0",
    "tasks": [
        {
            "label": "build-normal",
            "type": "shell",
            "command": "E:\\Projects\\SAMP\\compiler\\pawncc.exe",
            "args": ["\\\"${relativeFile}\\\"", "\\\"-;+\\\"", "\\\"-(+\\\"", "\\\"-d3\\\"", "\\\"-Z\\\""],
            "group": {
                "kind": "build",
                "isDefault": true
            },
            "isBackground": false,
            "presentation": {
                "reveal": "silent",
                "panel": "dedicated"
            },
            "problemMatcher": "$pawncc"
        }
    ]
}
```
The folder containing `pawncc.exe` is separated from my server files.

**EDIT N°1:** 
In the 'problems' tab, I noticed the path to the file in which the error is located is wrong. It has the following form:
```
E:\Projects\SAMP\gamemodes\servers\server-name\gamemodes\file.pwn
```
I guess this has something to do with the compiler being elsewhere, right?

**EDIT N°2:**
I was wondering why it put '/gamemodes/' before the relative path to the current file, and so I figured it must've been built-in the extension, and I believe I found the culprit:
https://github.com/Southclaws/vscode-pawn/blob/master/package.json#L65

**EDIT N°3:**
I changed the `problemMatcher` to `$sampctl` and that seemed to have gotten rid of the problem, however, it can still not find my modules. The paths seem correct now, though.

Also, I just read the README part where it says:

> problemMatcher is the part that allows recognising the Pawn compiler output and presenting it in the problems panel of the editor. This doesn't work well with external includes because the paths change from relative to absolute. sampctl fortunately fixes this (and a lot of other annoying things).   

I don't know what that means. I've looked into sampctl very briefly, but using it seems like a bit of a stretch. If it turns out that using sampctl is the only viable solution for now, then I will use it.

**EDIT N°4:**
http://forum.sa-mp.com/showpost.php?p=3976383&postcount=17 
This reply makes a lot more sense now that I have been going through the package files. I forked the repository and added another problemMatcher with only `${workspaceRoot}` and that didn't seem to fix it; as you have already established. Damn.

**Comment by Southclaws** [comment]
See #2 for some relevant information. It might help to use the `-D` flag for pawncc to specify the working directory. I'll have a proper read of this later and figure out a more comprehensive answer and/or fix for the issues with paths because I've noticed a few people running into this.

**Comment by Southclaws** [comment]
Okay so what's happening here is exactly the same as #2 - my assumption was wrong about the typical server folder setup. If you use the quick fix I provided in #2 you'll solve your issue and I will try to solve this on the extension end with minimal consequences.

sampctl is a different build toolchain that focuses on a more modern compilation workflow inspired by languages/ecosystems like Go, NodeJS and Ruby. It resolves a ton more issues than just this one and also facilitates proper package management and automatic plugin downloads.

In conclusion, it's clear that this setup is causing confusion for users so I'll try and improve the documentation and default configuration settings.

Since this is a duplicate of #2 I'll close this and push any further discussion there.
